moto/ec2/models/instance_types.py

```diff
@@ -17,12 +17,15 @@
 INSTANCE_TYPE_OFFERINGS: Dict[str, Any] = {}
 for _location_type in listdir(root / offerings_path):
     INSTANCE_TYPE_OFFERINGS[_location_type] = {}
-    for _region in listdir(root / offerings_path / _location_type):
-        full_path = offerings_path + "/" + _location_type + "/" + _region
+    for data_file in listdir(root / offerings_path / _location_type):
```

Comment:
In the previous iteration, `_region` would actually mean the filename, for example `us-east-1.json`. This rewrite has slightly more sensible variable-names, and also deals with both compressed and regular filenames (`us-east-1.json` and `us-east-1.json.gz`).